src/Microsoft.Health.Checkpoints/Storage/StorageCheckpointClient.cs

```diff
@@ -33,78 +33,74 @@ public StorageCheckpointClient(BlobContainerClient containerClient, ILogger<Stor
             _logger = logger;
         }
 
-        public virtual async Task<ICheckpoint> GetCheckpointAsync(string partition, string checkpointIdentifier)
+        public virtual async Task<ICheckpoint> GetCheckpointAsync(string partition, string checkpointIdentifier, CancellationToken token = default)
         {
             EnsureArg.IsNotNullOrEmpty(partition, nameof(partition));
             EnsureArg.IsNotNullOrEmpty(checkpointIdentifier, nameof(checkpointIdentifier));
 
-            ICheckpoint checkpoint = new Checkpoint();
+            var checkpoint = new Checkpoint();
+            var blobName = GetBlobName(partition, checkpointIdentifier);
 
-            try
+            IAsyncEnumerable<Page<BlobItem>> resultSegment = _storageClient.GetBlobsAsync(
+                                    traits: BlobTraits.Metadata,
+                                    states: BlobStates.All,
+                                    prefix: blobName,
+                                    cancellationToken: CancellationToken.None)
+                                    .AsPages();
+
+            await foreach (Page<BlobItem> blobPage in resultSegment)
             {
-                var resultSegment = _storageClient.GetBlobsAsync(
-                                        traits: BlobTraits.Metadata,
-                                        states: BlobStates.All,
-                                        prefix: $"{partition}/{checkpointIdentifier}",
-                                        cancellationToken: CancellationToken.None)
-                                        .AsPages();
-
-                await foreach (Page<BlobItem> blobPage in resultSegment)
+                if (blobPage.Values.Count == 0)
+                {
+                    _logger.LogInformation($"No blob found for blob name {blobName}");
```

Comment:
For structured logging to work, we should not use string interpolation. (Also elsewhere in this PR). The line should be something like:

```csharp
_logger.LogInformation("No blob found for blob name {blobName}", blobName);
```

Reply:
updated